src/Microsoft.AspNet.OData.Shared/ODataQueryContext.cs

```diff
@@ -72,18 +74,30 @@ public ODataQueryContext(IEdmModel model, IEdmType elementType, ODataPath path)
             {
                 throw Error.ArgumentNull("model");
             }
+
             if (elementType == null)
             {
                 throw Error.ArgumentNull("elementType");
             }
 
-            Model = model;
             ElementType = elementType;
+            Model = model;
             Path = path;
             NavigationSource = GetNavigationSource(Model, ElementType, path);
             GetPathContext();
         }
 
+        /// <summary>
+        /// Destructor called to clean up reference in ModelContainer
+        /// </summary>
+        ~ODataQueryContext()
```

Comment:
should we use the [IDisposable pattern](https://msdn.microsoft.com/en-us/library/system.idisposable(v=vs.110).aspx) instead of destructor?